generator.rb

```diff
@@ -0,0 +1,99 @@
+def prompt_for_input(which_word)
+  continue = false
+
+  until continue == true
+    print "Please enter the #{which_word} word: "
+    word = gets.chomp
+
+    until word.length > 1
+      print "Please enter a word with more than one character: "
+      word_a = gets.chomp
+    end
+
+    continue = validate(word)
+    if continue == false
+      puts "The word must contain at least one vowel."
+    end
+  end
+
+  return word
+end
+
+def is_vowel?(letter)
+  case letter
+  when "a", "e", "i", "o", "u", "A", "E", "I", "O", "U"
+    return true
+  else
+    return false
+  end
+end
+
+def find_vowels(word)
+  # create an array to hold the indices of all the vowels in word_a
+
+  vowel_array = []
+
+  #   find each vowel in the word and add its index to the array
+  # word.each_char.with_index { |c, i|
+  #   if is_vowel?(c)
+  #     vowel_array << i
+  #   end
+  # }
+
+  temp_array = word.split("")
+  temp_array.length.times do |i|
+    if is_vowel?(temp_array[i])
+      vowel_array << i
+    end
+  end
```

Comment:
While effective, this solution is doing a lot more work than it needs to. It will take the program less time to run if you just returned the index of the first or last value in the word.